test/test.go

```diff
@@ -50,6 +50,45 @@ func Addresses(n int) []string {
 	return addrs
 }
 
+// LocalHost returns the default local bind address: e.g. [::] or 127.0.0.1
+func LocalHost() string {
+	addr, err := n.ResolveTCPAddr("tcp", "localhost:0")
+	if err != nil {
+		panic(err)
+	}
+	return addr.IP.String()
+}
+
+// FreeBind provides an address for binding on provided address
+func FreeBind(a string) string {
```

Comment:
This method was already somewhere before no ? Why copy it ?

Reply:
we have free port. this one is more general in that it allows specification and returns the host part. so e.g. I can start with 'localhost' and learn if that translated into '127.0.0.1' or '[::1]', or i can start with '0.0.0.0' and learn that that translated into an external IP that i can then make a ssl cert for.

Is it worth finding all the uses of `FreePort` and having it either use this or refactoring the client to use this instead?